test/isolation/specs/multi_transaction_indexing.spec

```diff
@@ -0,0 +1,116 @@
+
+setup {
+    CREATE TABLE ts_index_test(time int, temp float, location int);
+    SELECT create_hypertable('ts_index_test', 'time', chunk_time_interval => 10, create_default_indexes => false);
+    INSERT INTO ts_index_test VALUES (1, 23.4, 1),
+        (11, 21.3, 2),
+        (21, 19.5, 3);
+
+    CREATE TABLE barrier(i INTEGER);
+}
+
+teardown {
+    DROP TABLE ts_index_test;
+    DROP TABLE barrier;
+}
+
+session "CREATE INDEX"
+step "F" { SET client_min_messages TO 'fatal'; }
+step "CI"	{ CREATE INDEX test_index ON ts_index_test(location) WITH (hypertable.multi_transaction, hypertable.barrier_table='barrier'); }
+
+session "RELEASE BARRIER"
+setup		{ BEGIN; SET LOCAL lock_timeout = '50ms'; SET LOCAL deadlock_timeout = '10ms'; LOCK TABLE barrier;}
+step "Bc"   { ROLLBACK; }
+
+session "SELECT"
+setup		{ BEGIN; SET LOCAL lock_timeout = '50ms'; SET LOCAL deadlock_timeout = '10ms'; }
+step "S1"	{ SELECT * FROM ts_index_test; }
+step "Sc"	{ COMMIT; }
+
+session "INSERT CHUNK"
+setup		{ BEGIN; SET LOCAL lock_timeout = '50ms'; SET LOCAL deadlock_timeout = '10ms'; }
+step "I1"	{ INSERT INTO ts_index_test VALUES (31, 6.4, 1); }
+step "Ic"	{ COMMIT; }
+
+session "DROP INDEX"
+step "DI"	{ DROP INDEX test_index; }
+
+session "RENAME COLUMN"
+step "RI"	{ ALTER TABLE test_index RENAME COLUMN location TO height; }
+
+session "COUNT INDEXES"
+step "P"    {  SELECT index_size FROM timescaledb_information.hypertable; }
+
+# we need to COMMIT every transaction started in setup regardless of whether we use them
+# inserts work between chunks
+permutation "CI" "I1" "Ic" "Bc" "P" "Sc"
+
+# create blocks on insert
+permutation "I1" "CI" "Bc" "Ic" "P" "Sc"
+
+# create blocks on select
+permutation "S1" "CI" "Bc" "Sc" "P" "Ic"
+
+# drop works (the error message outputs an OID, remove the "F" to see the error)
+permutation "F" "CI" "DI" "Bc" "P" "Ic" "Sc"
+
+# rename should block
+permutation "CI" "RI" "Bc" "P" "Ic" "Sc"
+
+# Ideally we would declare these functions in setup, and use them to check that the actual index
```

Comment:
can't we separate these 2 functions into 2 sql statements? Or do they still not fit?

Reply:
is there something like a runner.sh for isolation tests we can stick this in?

Reply:
1. Unfortunately, I think it counts the entire `setup` block as one statement. I can try adding a new `session` to define them in 🤔 
2. The runner thing might work, though isolationregress handles the runner differently than the regular one, so I'll need to check.